src/libraries/System.Runtime.InteropServices.JavaScript/gen/JSImportGenerator/Marshaling/BaseJSGenerator.cs

```diff
@@ -10,62 +10,48 @@
 
 namespace Microsoft.Interop.JavaScript
 {
-    internal abstract class BaseJSGenerator : IJSMarshallingGenerator
+    internal abstract class BaseJSGenerator(TypePositionInfo info, StubCodeContext codeContext) : IBoundMarshallingGenerator
     {
-        protected IBoundMarshallingGenerator _inner;
-        public MarshalerType Type;
+        private static ValueTypeInfo JSMarshalerArgument = new ValueTypeInfo(Constants.JSMarshalerArgumentGlobal, Constants.JSMarshalerArgument, IsByRefLike: false);
 
-        protected BaseJSGenerator(MarshalerType marshalerType, IBoundMarshallingGenerator inner)
```

Comment:
As we haven't implemented the nesting support (using a non-JS generator followed by a JS generator) and we aren't doing a ton of investment in this area, I've removed most of the infrastructure for it.

In the future if we add support, we can add it back as a subclass of this type at that time instead of carrying the infrastructure without a use case until then.

Reply:
I'm not sure what you mean by "nesting support" but I would love to do nested generic types one nice day in the future. 
Like `Func<Task<Nullable<int,string>>> param1` . 
That would need us to collect such type information and as well generate nested lambdas.

But it's not top priority and maybe your refactorings are not really make it more difficult.

Reply:
By "nesting support" I meant the idea of using a `CustomMarshallerAttribute`-style marshaller to marshal a type to a C-style representation and then using a JSMarshalAs marshaller on that value to convert it to a JS representation. We spoke about this when you were first building up the JSImport/JSExport generators, but we never got to actually doing anything with it.

Reply:
I see, thanks